one_liners2.cc

```diff
@@ -0,0 +1,56 @@
+#include <algorithm>
+#include <iostream>
+#include <numeric>
+#include <vector>
+#include <string>
+#include <boost/algorithm/string/classification.hpp>
+
+using namespace std;
+
+// 1. Xor to each char
+string xored_string(string s, char n) {
+    return accumulate(s.begin(), s.end(), string(""), [n](
+        const string& acc, const char& c){return acc + char(c^n);});
+}
+// reverse
+string my_reverse(string s) {
+    return accumulate(s.crbegin(), s.crend(), string(""), [](
+        const string& acc, const char& c){return acc + c;});
+}
+
+// check all capitals
+bool all_capitals(const vector<string>& names) {
+    return accumulate(names.begin(), names.end(), true, [](
+        bool acc, const string& name){
+            return acc & 
```

Comment:
You probably mean `&&`  (BTW, what would be the difference?)

Reply:
yeah, it is logical 'and' not bitwise.  Does the compiler converts the booleans to numeric to perform the  bitwise &?
Practically, I believe that although it is weird, it should work, as long as it is mapped true->1 and false->0.
I will fix, of course.

Reply:
Yes, in this case _integral promotion_ takes place (see C++20 standard, 7.3.6).  The problem is, however, that bitwise operations are _fully_ evaluated, while the logical ones have _short-circuit_ evaluation. So this is a nice example of how a single symbol might significantly affect performance :).

Reply:
Hmmm,  I see!